File: citizen_science/lib/dto/register_request.dart
/// Data transfer object for user registration requests.
/// 
/// Contains all required information for creating a new user account.
class RegisterRequest {
  final String nome;
  final String cognome;
  final String email;
  final String password;
  final String ruolo;

  RegisterRequest({
    required this.nome,
    required this.cognome,
    required this.email,
    required this.password,
    this.ruolo = 'utente',
  });

  /// Converts this request to a JSON map for API transmission.
  Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'cognome': cognome,
      'email': email,
      'password': password,
      'ruolo': ruolo,
    };
  }
}

File: CitizenScienceBackend/src/main/java/com/citizenScience/dto/RegisterRequest.java
package com.citizenscience.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Data Transfer Object for user registration requests.
 * Contains user details and validation constraints.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RegisterRequest {
    @NotBlank(message = "Nome is required")
    private String nome;

    @NotBlank(message = "Cognome is required")
    private String cognome;

    @NotBlank(message = "Email is required")
    @Email(message = "Email should be valid")
    private String email;

    @NotBlank(message = "Password is required")
    private String password;

    private String ruolo = "utente";
}
